charts/skypilot/templates/api-deployment.yaml

```diff
@@ -154,6 +178,13 @@ spec:
       - name: state-volume
         emptyDir: {}
       {{- end }}
+      {{- if .Values.kubernetesCredentials.useKubeconfig }}
+      - name: kube-config
+        secret:
+          secretName: kube-credentials
+      - name: kube-config-store
+        emptyDir: {}
```

Comment:
I'm bit concerned about this indirection:

- The copied kubeconfig is detached from the secret thus changes to secret won't be reflected to API server as before;
- Changes to the copied kubeconfig are lost when API server pod gets recreated;

, which are contrary to the immutable infrastructure principle in some ways, adding variables to problem reasoning.

How about support specifying the name of kubeconfig secret so that users can copy the original secret and make changes to the copied one persistently?

Reply:
Thanks @aylei, that's a good point. I didn't realize mounted secrets are auto reloaded. I have updated the PR to just allow specifying name of the secret to use.